src/BuiltInTools/dotnet-watch/README.md

```diff
@@ -91,6 +91,91 @@ Example:
   </ItemGroup>
 ```
 
+### Hot Reload
+
+Starting in .NET 6, dotnet-watch adds support for hot reload. Hot reload is a set of technologies that allows applying changes to a running app without having to rebuild it. This enables a much faster development experience as developers receive immediate feedback when modifying their apps during local development.
+
+When a file is modified, watch determines if it can be hot reloaded. Here's how different kinds of files are handled:
```

Comment:
I think it makes sense to call out the fact that the implementation does have some non-trivial logic for determining the executable to run/apply deltas to.

Also maybe document why we do `dotnet exec App.dll` instead of `dotnet run`? As in, we want to avoid spawning multiple dotnet processes and want to be able to pass a value for startup hooks.